subsys/lorawan/lorawan.c

```diff
@@ -146,29 +143,92 @@ const char *eventinfo2str(int status)
 	}
 }
 
+/*
+ * MAC status and Event status to Zephyr error code conversion.
+ * Direct mapping is not possible as statuses often indicate the domain from
+ * which the error originated rather than its cause or meaning. -EINVAL has been
+ * used as a general error code because those usually result from incorrect
+ * configuration.
+ */
+const int mac_status_to_errno[] = {
```

Comment:
I don't particularly like these sort of lists myself since relying on the order can be a bit error prone. I generally prefer using designated initizlizers (e.g., [LORAMAC_STATUS_BUSY=-EBUYS]). I have certainly seen it in Linux, but I don't know if it is customary in Zephyr. Some grepping suggests that it is used though.

Reply:
> Some grepping suggests that it is used though.

In networking subsystem these kind of arrays are not used as they are error prone.

Reply:
Why would it be more error prone than anything else? The only two drawbacks compared to a switch statement seems to be that you don't get bounds checking for free and that holes get mapped to zero. Both could easily be implemented by using a lookup function that checks the bound and that the returned value isn't inadvertently mapped to zero. The benefit is that you are guaranteed a table that is lives in flash whereas a switch statement could become anything (some embedded compilers generate code that sequentially scans a generated branch table, others just have a gazillion comparisons).